### PR TITLE
Improve rosout logging

### LINKED ISSUE #38: Improve rosout/logging output
Currently there is either way too much debug prints, or too little.
Improve the logging/console output so that the default output is useful without being too verbose.

### PR DESCRIPTION
**Pull request by icolwell-as**
Resolves #38 by improving the rosout logging.

- Make use of ROS_DEBUG API instead of a special debug_print parameter.
- Improve the output that is generated from configuration and re-configuration via dynamic reconfigure.
- Improve the output when the vimba system starts up and looks for cameras.
- Re-arrange and refactor a bit of related code while I was there.
- Remove code/output that isn't very useful.

### CODE REVIEW

File: include/avt_vimba_camera/avt_vimba_api.h [hunk 1]
```diff
@@ -100,6 +100,33 @@ class AvtVimbaApi {
...
+      case VmbInterfaceUsb: return "USB";
+        break;
+      default: return "Unknown";
+    }
+  }
+
+  std::string accessModeToString(VmbAccessModeType modeType) {
+    std::string s;
+    if (modeType & VmbAccessModeFull)
+      s = std::string("Read and write access");
```
Comment (jilinzhouas) [inline include/avt_vimba_camera/avt_vimba_api.h]: How about `return \"Read and write access\"` so it matches `interfaceToString` function?

File: include/avt_vimba_camera/avt_vimba_api.h [hunk 2]
```diff
@@ -171,26 +198,25 @@ class AvtVimbaApi {
   VimbaSystem& vs;
 
   void listAvailableCameras(void) {
-    std::string name;
+    ROS_INFO("Searching for cameras ...");
     CameraPtrVector cameras;
     if (VmbErrorSuccess == vs.Startup()) {
       if (VmbErrorSuccess == vs.GetCameras(cameras)) {
-        for (CameraPtrVector::iterator iter = cameras.begin();
```
Comment (jilinzhouas) [inline include/avt_vimba_camera/avt_vimba_api.h:178]: how about `for (const auto& camera : cameras)`?

File: include/avt_vimba_camera/avt_vimba_camera.h [hunk 3]
```diff
@@ -142,7 +141,7 @@ class AvtVimbaCamera {
   std::string trigger_source_;
   int trigger_source_int_;
 
-  CameraPtr openCamera(std::string id_str);
+  CameraPtr openCamera(std::string id_str, bool print_all_features);
```
Comment (jilinzhouas) [inline include/avt_vimba_camera/avt_vimba_camera.h]: const std::string&

File: include/avt_vimba_camera/avt_vimba_api.h (hunk 1 above)
Comment (icolwell-as) [reply include/avt_vimba_camera/avt_vimba_api.h]: Sounds good! this function was just moved from another file, I'll improve it though.

**Comment by dvangeyn-as** [comment]
Much cleaner to read now. Appreciate that.

**Review by dvangeyn-as — APPROVED** [review]

**Review by jilinzhouas — APPROVED** [review]

**Pull request merged**